pkg/registry/core/pod/strategy_test.go

```diff
@@ -82,6 +82,20 @@ func TestMatchPod(t *testing.T) {
 			fieldSelector: fields.ParseSelectorOrDie("status.phase=Pending"),
 			expectMatch:   false,
 		},
+		{
+			in: &api.Pod{
+				Spec: api.PodSpec{SchedulerName: "custom-scheduler"},
+			},
+			fieldSelector: fields.ParseSelectorOrDie("spec.schedulerName=custom-scheduler"),
+			expectMatch:   true,
+		},
+		{
+			in: &api.Pod{
+				Spec: api.PodSpec{SchedulerName: "default-scheduler"},
+			},
+			fieldSelector: fields.ParseSelectorOrDie("spec.schedulerName=custom-scheduler"),
```

Comment:
This is just a negative test and trying to list which does not exist, this value can be anything in fact.